### PR TITLE
DateTimeUtil refactored

### PR DESCRIPTION
**Pull request by nikitasidorevich**

### CODE REVIEW

File: sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/util/DateTimeUtil.java [hunk 1]
```diff
@@ -25,58 +25,48 @@
...
+     * @param sc search criteria to be actualized
+     */
     public static void actualizeSearchCriteriaDate(DateSearchCriteria sc) {
-        if (sc.getDate() != null) {
-            if (isToday(sc.getDate())) {
-                sc.setDate(formatDateToStartOfDay(new Date()));
-            } else {
-                sc.setDate(formatDateToStartOfDay(sc.getDate()));
-            }
+        Date date = sc.getDate();
```
Comment (nikitasidorevich) [inline sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/util/DateTimeUtil.java]: `isToday` check was useless - we force set `searchCriteria` date to today's date start of day regardless of it's result.

Comment (tsvrko) [reply sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/util/DateTimeUtil.java]: No, it used to format date to passed in criteria day's start. With this changes you forcibly show user only today's runs.

Comment (nikitasidorevich) [reply sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/util/DateTimeUtil.java]:
```
if (sc.getDate() != null) {
    if (isToday(sc.getDate())) {
        sc.setDate(formatDateToStartOfDay(new Date()));
    } else {
        sc.setDate(formatDateToStartOfDay(sc.getDate()));
    }
}
```
Good catch. I actually got if condition wrong. Ok, so it seems what in really going on is this:
- if date is not null we check if date is actually today (current date):
  a. if yes - we re-create date (with new Date()) which is actaully today's date + current time and format it to start day: e.g. Aug 23, 2019 15:22 -> Aug 23, 2019 00:00
  b. if no - we use date supplied and reset it to start of day.

I don't see why do we need `isToday` check anyway. Bottom line - we reset date to start of day regardless of fact if it's today's date or not. Will update to:
```
    /**
     * Force set search criteria date to start of day
     * @param sc search criteria to be actualized
     */
    public static void actualizeSearchCriteriaDate(DateSearchCriteria sc) {
        Date date = sc.getDate();
        if (date != null) {
            sc.setDate(toStartOfDay(date);
        }
    }
```

Comment (tsvrko) [reply sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/util/DateTimeUtil.java]: It's ok now, *isToday* was redundant check.

**Pull request merged**